cms/djangoapps/contentstore/tests/test_course_listing.py

```diff
@@ -107,6 +109,88 @@ def test_errored_course_global_staff(self):
             courses_list_by_groups, __ = _accessible_courses_list_from_groups(self.request)
             self.assertEqual(courses_list_by_groups, [])
 
+    @ddt.data(ModuleStoreEnum.Type.split, ModuleStoreEnum.Type.mongo)
+    def test_staff_course_listing_performance(self, default_store):
```

Comment:
I'm not crazy about timing how long it would take. Instead assert how many mongo queries are run.

Secondly, why is this only a performance improvement for global staff? Can we make this work for anyone accessing the studio index page, or is that not straightforward?